starlite/utils/extractors.py

```diff
@@ -0,0 +1,416 @@
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Coroutine,
+    Dict,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
+
+from starlette.requests import cookie_parser
+from typing_extensions import Literal, TypedDict
+
+from starlite.connection import Request
+from starlite.datastructures import UploadFile
+from starlite.enums import HttpMethod, RequestEncodingType
+
+if TYPE_CHECKING:
+    from starlite.connection import ASGIConnection
+    from starlite.types import Method
+    from starlite.types.asgi_types import HTTPResponseBodyEvent, HTTPResponseStartEvent
+
+
+def obfuscate(values: Dict[str, Any], fields_to_obfuscate: Set[str]) -> Dict[str, Any]:
+    """
+    Args:
+        values: A dictionary of strings
+        fields_to_obfuscate: keys to obfuscate
+
+    Returns:
+        A dictionary with obfuscated strings
+
+    """
+    if not fields_to_obfuscate:
+        return values
+
+    output = {}
+    for key, value in values.items():
+        if any(field.lower() == key.lower() for field in fields_to_obfuscate):
+            output[key] = "*****"
+        else:
+            output[key] = value
+    return output
+
+
+RequestExtractorField = Literal[
```

Comment:
Why are we using a Literal in preference to an enum?

Reply:
because this very same value is user by users in the config. While enums are great, they are somewhat cumbersum when you need to define somethin like this:

```python
request_logging_config = RequestLoggingConfig(request_log_fields=(
        "path",
        "method",
        "query",
        "path_params",
        "body",
    ))
```